File: ShadowFrameworkLite20/src/shadow/pipeline/expression/SFExpressionTypeWrapper.java
package shadow.pipeline.expression;

public class SFExpressionTypeWrapper extends SFExpressionElement{

	public SFExpressionTypeWrapper(short type) {
		super("wr"+type);
		setType(type);
	}
	
	public SFExpressionTypeWrapper(short type,SFExpressionElement wrappedExpression) {
		super("wr"+type);
		setType(type);
		addSubExpression(wrappedExpression);
	}

	@Override
	public void addSubExpression(SFExpressionElement element)
			throws ArrayIndexOutOfBoundsException {
		list.add(element);
	}
	
	@Override
	public void evaluateType() {
		//nothing to do.
	}
}

File: trunk/ShadowFrameworkJS/SF20Core/src/shadow/pipeline/expression/SFExpressionTypeWrapper.js

function SFExpressionTypeWrapper(type){
	super("wr"+type);//Warning: Not well Identified 
	setType(type);//Warning: Not well Identified 
}
function SFExpressionTypeWrapper(type, wrappedExpression){
	super("wr"+type);//Warning: Not well Identified 
	setType(type);//Warning: Not well Identified 
	addSubExpression(wrappedExpression);//Warning: Not well Identified 
}

SFExpressionTypeWrapper.prototype = {
};